src/coreclr/jit/importer.cpp

```diff
@@ -7562,8 +7562,8 @@ void Compiler::impImportBlockCode(BasicBlock* block)
             case CEE_BRFALSE_S:
 
                 /* Pop the comparand (now there's a neat term) from the stack */
-                // TODO: we might want to only perform gtFoldExprConst here for TP
-                op1 = gtFoldExpr(impPopStack().val);
+                op1 = impPopStack().val;
+                op1 = opts.OptimizationEnabled() ? gtFoldExpr(op1) : op1;
```

Comment:
Is it worth having a helper `impPopStackAndFoldExpr()`?

We can then mark it as `forceinline` and have it do the two step thing + relevant optimization checks?

Reply:
Alternatively, should we just have `foldExpr` be a more general part of `impPushStack(node)`?

We already do explicit `gtFoldExpr` as part of pushing several node types up, but not more generally.

Reply:
Probably? I assume it needs to be justified in other places as in general it could be a TP regression to run gtFoldExpr where it won't fold anything. But in this place it makes sense to do because BRTRUE/BRFALSE are branches + condition so we potentially remove a branch early

Reply:
This is similar to what we already do for other branches so seems fine.

Doing this more broadly, as Egor says, may not accomplish much other than make the jit slower. But it's probably worth prototyping.